repository.go

```diff
@@ -83,7 +83,7 @@ func (r *Repository) GetPowerSources() ([]*PowerSource, error) {
 func (r *Repository) GetPowerSource(id string) (*PowerSource, error) {
 	nid, err := strconv.ParseInt(id, 10, 32)
 	if err != nil {
-		return nil, ErrNotFound
+		return nil, NewErrBadRequestF("id is bad '%v'", id)
```

Comment:
It's kind of an philosophical issue, but I'd expect a `404` instead of a `400`. The request is perfectly fine, the resource just doesn't exist.

Reply:
Shall I remove replace `NewErrBadRequestF("id is bad '%v'", id)` with something like `NewErrNotFoundF("id is bad '%v'", id)` or revert this return errror change?

Reply:
Do you have slack or something else? I would like to discuss some other upcoming changes with you.

Reply:
> Shall I replace `NewErrBadRequestF("id is bad '%v'", id)` with something like `NewErrNotFoundF("id is bad '%v'", id)` or revert this return errror change?

Both would be fine with me.

Reply:
> Do you have slack or something else? I would like to discuss some other upcoming changes with you.

Unfortunately, our slack isn't open. Just create an issue if you want feedback on something.

Reply:
change it

ok, I will open a new issue.